### PR TITLE
PMC Submitter

### PR DESCRIPTION
**Pull request by jhoanmanuelms**
- Add a new mode to submit the imported submissions
- Retrieve and deserialize all the imported submissions
- Create a separated module for infrastructure related tasks
- Add scripts and Gradle task to create a local database
- Rename some classes and packages in order to keep consistency

TODO
- Use security service to authenticate and perform the submission instead of using the token directly
- Make a submission for a study with files

### CODE REVIEW

File: scheduler/pmc-processor-task/src/main/kotlin/ac/uk/ebi/pmc/data/MongoRepository.kt [hunk 1]
```diff
@@ -37,4 +46,11 @@ class MongoRepository(
         collection.insertOne(FileDoc(file.name, file.absolutePath, accNo))
         return collection.findOne(Filters.eq("path", file.absolutePath))!!.id
     }
+
+    suspend fun getSubFiles(ids: List<ObjectId>): MutableList<Option<FileDoc>> {
```
Comment (Juan-EBI) [inline scheduler/pmc-processor-task/src/main/kotlin/ac/uk/ebi/pmc/data/MongoRepository.kt]: why a mutable list? I do not see any scenario that we need to update the list of files so better keep it inmutable

Comment (Juan-EBI) [inline scheduler/pmc-processor-task/src/main/kotlin/ac/uk/ebi/pmc/data/MongoRepository.kt]: I think we should not retrieve a list of options , I think there must be a method that can be used when we are sure element exists

File: infrastructure/src/main/resources/setup/database/schema.sql [hunk 2]
```diff
@@ -0,0 +1,284 @@
+CREATE TABLE AccessTag (
+    id            BIGINT AUTO_INCREMENT PRIMARY KEY,
+    description   VARCHAR(255) NULL,
+    name          VARCHAR(255) NULL,
+    owner_id      BIGINT       NULL,
+    parent_tag_id BIGINT       NULL,
+    CONSTRAINT access_tag_name_idx UNIQUE (name),
+    CONSTRAINT FK5q8ww38t2jsgfy73t0e12am52 FOREIGN KEY (parent_tag_id) REFERENCES AccessTag (id)
+);
```
Comment (Juan-EBI) [inline infrastructure/src/main/resources/setup/database/schema.sql]: I think `owner_id` and `parent_tag_id` are not required

File: scheduler/pmc-processor-task/src/main/kotlin/ac/uk/ebi/pmc/submit/PmcSubmitter.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    private val submissionDocService: MongoDocService,
+    private val serializationService: SerializationService
+) {
+    suspend fun submit() =
+        submissionDocService.getAllSubmissions().map { GlobalScope.launch { processSubmission(it) } }
+
+    private suspend fun processSubmission(submission: SubmissionDoc) = coroutineScope {
+        val deserialized = serializationService.deserializeSubmission(submission.body.replace("\\", ""), SubFormat.JSON)
+        logger.info { "to submit ${deserialized.accNo}" }
+        val response = bioWebClient.submitSingle(submission.body.replace("\\", ""), SubmissionFormat.JSON)
```
[the remark this answers is not in the document]
Comment (jhoanmanuelms) [reply scheduler/pmc-processor-task/src/main/kotlin/ac/uk/ebi/pmc/submit/PmcSubmitter.kt]: The first one is actually not required, it was supposed to be a test. I'll just remove it.

File: scheduler/pmc-processor-task/src/main/kotlin/ac/uk/ebi/pmc/data/MongoRepository.kt (hunk 1 above)
Comment (jhoanmanuelms) [reply scheduler/pmc-processor-task/src/main/kotlin/ac/uk/ebi/pmc/data/MongoRepository.kt]: Done

File: infrastructure/src/main/resources/setup/database/schema.sql (hunk 2 above)
Comment (jhoanmanuelms) [reply infrastructure/src/main/resources/setup/database/schema.sql]: 👍

File: scheduler/pmc-processor-task/src/main/kotlin/ac/uk/ebi/pmc/data/MongoRepository.kt (hunk 1 above)
Comment (jhoanmanuelms) [reply scheduler/pmc-processor-task/src/main/kotlin/ac/uk/ebi/pmc/data/MongoRepository.kt]: Fixed

**Pull request merged**